### ISSUE TITLE
Tooltip can't find elements with Invalid Selectors as IDs

### ISSUE THREAD

**Issue by explorigin** [rackerlabs/helix-ui#303]
## Describe the Bug
Tooltip fails whenever the target id makes an invalid CSS selector.

## Steps to Reproduce

1. Create a trigger element like so:
```
<span data-tooltip="2d90208e58f24216be64de517570c628">Hello</span>
<hx-tooltip id="2d90208e58f24216be64de517570c628">Do you like my hat?</hx-tooltip>
```
2. Hover over **Hello** with your mouse.
3. Observe error (screenshotted below) in the dev console.

## Expected behavior
I expected it to ask me about my hat.

## Screenshots
<img width="1030" alt="image" src="https://user-images.githubusercontent.com/697818/43796358-aa37933a-9a49-11e8-9053-c56529b51736.png">

## Environment
Please complete the following information:

 - Device: Lappy
 - OS: OSX
 - Browser: Chrome 68

## Additional Context
Problem is [here](https://github.com/rackerlabs/helix-ui/blob/master/src/helix-ui/elements/HXTooltipElement.js#L58).  Make it look like:
```js
this._target = this.getRootNode().querySelector('[data-tooltip="' + this.id + '"]');
```

Proof that the fix works:
<img width="1043" alt="image" src="https://user-images.githubusercontent.com/697818/43796187-2820fc2e-9a49-11e8-9c58-80611cb1a95e.png">

**Comment by CITguy** [comment]
Is the ID being generated by a library or by your own business logic?

The logic in the assignment of `this._target` works for valid IDs. If an invalid ID is being generated, we can't do much about it in HelixUI.

**Comment by explorigin** [comment]
This is about removing warts from Helix.  Yes, you can always prepend a letter to make it valid, but who wants to have to do that (or figure out that you have to)?  A library should just work...and the fix to accommodate it is dead simple.

**Comment by explorigin** [comment]
Technically the id is valid HTML but it makes for an invalid CSS selector...hence the need for quotes.

**Comment by CITguy** [comment]
I apologize. 

I confused the snippet above for `this._target = ...` as _current_ code when you were actually providing _corrected_ code. 

Thank you for reporting this bug. 

We'll get it fixed!

-----

**Additional Information**

Demo of Bug: https://codepen.io/CITguy/pen/GBYWYb

[W3C spec](https://www.w3.org/TR/html50/dom.html#the-id-attribute) allows alphanumeric ID values (including `_`, `-`, and `.`), so the mentioned ID (`2d90208e58f24216be64de517570c628`) is a valid HTML5 value.

**Comment by CITguy** [comment]
@explorigin: now fixed in `helix-ui@0.12.1`
